include/deal.II/lac/linear_operator.h

```diff
@@ -769,6 +769,259 @@ null_operator(const LinearOperator<Range, Domain> &op)
+ * \f$ \mathbf{K} \f$ and, similarly, general subvectors of
+ * \f$ \mathbf{d},\mathbf{h} \f$ are given by \f$ x,y,f,g \f$ .
+ *
+ * This is equivalent to the following two statements:
+ * @f{eqnarray*}
+     (1) \quad Ax + By &=& f \\
+     (2) \quad Cx + Dy &=& g \quad .
+ * @f}
+ *
+ * Assuming that \f$ A,D \f$ are both square and invertible, we could
+ * then perform one of two possible substitutions.
+ * @f{eqnarray*}
+     (3) \quad x &=& A^{-1}(f - By) \quad \textnormal{from} \quad (1) \\
+     (4) \quad y &=& D^{-1}(g - Cx) \quad \textnormal{from} \quad (4) .
+ * @f}
+ *
+ * For the purpose of the current implementation, we choose to
+ * substitute (3) into (2):
+ * @f{eqnarray*}
+     (5) \quad C \: A^{-1}(f - By) + Dy &=& g \\
+     (6) \quad -C \: A^{-1} \: By + Dy &=& g - C \: A^{-1} \: f
+   @f}
+ * which leads to the result
+ * @f[
+     (7) \quad Sy = g - C \: A^{-1} f
+   @f]
+ * with \f$ S = (D - C\: A^{-1} \:B) \f$ being the Schur complement.
+ * Note that for this choice of \f$ S \f$, submatrix \f$ D \f$
+ * need not be invertible and may thus be the null matrix.
+ *
+ * So for any arbitrary vector \f$ a \f$, the Schur complement
+ * performs the following operation:
+ * @f[
+     (8) \quad Sa = (D - C \: A^{-1} \: B)a
+   @f]
+ *
+ * A typical set of steps needed the solve a linear system (1),(2)
+ * would be:
+ * 1. Define the inverse matrix @p A_inv (using inverse_operator()).
+ * 2. Define the Schur operator \f$ S \f$ (using schur_operator()).
+ * 3. Define iterative inverse matrix \f$ S^{-1} \f$ such that (8)
+ *    holds.
+ *    It is necessary to use a solver with a preconditioner
+ *    to compute the approximate inverse operation of \f$ S \f$ since
+ *    we never compute \f$ S \f$ directly, but rather the result of
+ *    its operation.
+ *    To achieve this, one may again use the inverse_operator() in
+ *    conjunction with the Schur operator that we've just
+ *    constructed.
+ * 4. Perform pre-processing step on the RHS of (7) using
+ *    preprocess():
+ *    @f[
+        g' = g - C \: A^{-1} \: f
+      @f]
+ * 5. Solve for \f$ y \f$ in (7):
+ *    @f[
+        y =  S^{-1} g'
+      @f]
+ * 6. Perform the post-processing step from (3) using
+ *    postprocess():
+ *    @f[
+        x =  A^{-1} (f - By)
+      @f]
+ *    Note that, at the same time, this function will reverse
+ *    the condensation operation performed in step (4) so that
+ *    the original RHS remains:
+ *    @f[
+        g = g' + C \: A^{-1} \: f
+      @f]
+ *
+ * @note The functionality provided by solve_system() encapulates
+ * all of the logic required to solve a block system using the
+ * Schur complement approach.
+ *
+ * @see
+ * @ref GlossBlockLA "Block (linear algebra)"
+ * @author Jean-Paul Pelteret, 2015
+ *
+ * @ingroup LAOperators
+ */
+template <typename Range_1, typename Domain_1,
+          typename Range_2, typename Domain_2>
+LinearOperator<Range_2, Domain_2>
+schur_operator(const LinearOperator<Range_1, Domain_1> &A_inv,
```

Comment:
without having looked at the rest of the patch, since you're creating an object that represents the Schur *complement*, any particular reason to call the function schur_*operator*?